### ISSUE TITLE
MS writing with legacy array shape fails

### ISSUE THREAD

**Issue by telegraphic** [RadioAstronomySoftwareGroup/pyuvdata#1379]
On main branch, commit [7790d06](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/commit/7790d0650a6407bd2d59277a208c92471dad2e52), MS writing fails if `use_future_array_shapes=False` is set.

### Minimal example

```python
from pyuvdata.data import DATA_PATH
from pyuvdata import UVData
import os

testfile = os.path.join(DATA_PATH, "day2_TDEM0003_10s_norx_1src_1spw.ms")

# This runs
uvobj = UVData()
uvobj.read(testfile, use_future_array_shapes=True)
uvobj.write_ms("temp_future_array_shapes.ms")

# This fails
uvobj = UVData()
uvobj.read(testfile, use_future_array_shapes=False)
uvobj.write_ms("temp_past_array_shapes.ms")
```

Will issue a traceback:

```
Traceback (most recent call last):
  File "/Users/daniel.price/Data/pyuvdata/temp_test.py", line 13, in <module>
    uvobj.write_ms("temp2.ms")
  File "/Users/daniel.price/Data/pyuvdata/pyuvdata/uvdata/uvdata.py", line 12804, in write_ms
    ms_obj.write_ms(
  File "/Users/daniel.price/Data/pyuvdata/pyuvdata/uvdata/ms.py", line 1219, in write_ms
    ms.putcol(col, temp_vals)
  File "/Users/daniel.price/local/mamba/envs/aa3/lib/python3.11/site-packages/casacore/tables/table.py", line 1155, in putcol
    self._putcol(columnname, startrow, nrow, rowincr, value)
RuntimeError: ArrayColumn::putColumn - column DATA has 1360, array has 4 rows: Table array conformance error
```

### Culprit and suggested fix

Relevant code is [L1204-1219](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/7790d0650a6407bd2d59277a208c92471dad2e52/pyuvdata/uvdata/ms.py#L1204C1-L1220C1), and was introduced in https://github.com/RadioAstronomySoftwareGroup/pyuvdata/commit/3a0bcbdfc54b2a997598e8160f9a3477de315809

```
        # Add all the rows we need up front, which will allow us to fill the
        # columns all in one shot.
        ms.addrows(self.Nblts * self.Nspws)
        if self.Nspws == 1:
            # If we only have one spectral window, there is nothing we need to worry
            # about ordering, so just write the data-related arrays as is to disk
            for attr, col in zip(attr_list, col_list):
                if self.future_array_shapes:
                    temp_vals = getattr(self, attr)[:, :, pol_order]
                else:
                    temp_vals = getattr(self, attr)[:, 0, :, pol_order]

                if flip_conj and (attr == "data_array"):
                    temp_vals = np.conj(temp_vals)

                ms.putcol(col, temp_vals)
```

For the test data, the input shape is `(1360, 1, 64, 4)`, but it comes out as `(4, 1360, 64)`! 

This seems to be a numpy fancy indexing quirk, and I certainly found it unexpected:

```python
import numpy as np
pol_order = [0, 2, 3, 1]         # Pol order from the minimal example above
d = np.zeros((1360, 1, 64, 4))   # Input data shape from the minimal example above
d_out = d[:, 0, :, pol_order]
print(d_out.shape)     # Prints (4, 1360, 64), unexpected!

d_out = d[:, 0, :][..., pol_order]
print(d_out.shape)     # Prints (1360, 64, 4)
```

The simple fix is to change

```python
temp_vals = getattr(self, attr)[:, 0, :, pol_order]
```
to 
```python
temp_vals = getattr(self, attr)[:, 0, :][..., pol_order]
```

If anyone can explain the numpy indexing quirk, I'd be happy to hear 🤔 .

**Comment by telegraphic** [comment]
Just wanted to add this fun example:

```python
>>> d_out = d[:, :, :, [0, 1]]
>>> d_out.shape
(1360, 1, 64, 2)
>>> d_out = d[:, 0, :, [0, 1]]
>>> d_out.shape
(2, 1360, 64)
```

**Comment by telegraphic** [comment]
Should now be fixed in #1388

**Comment by kartographer** [comment]
Fixed in #1392 (which was a rebased version of #1388)
